dask/array/tests/test_routines.py

```diff
@@ -2565,15 +2565,34 @@ def test_average(a, returned):
     assert_eq(np_avg, da_avg)
 
 
-def test_average_weights():
+@pytest.mark.parametrize("a", [np.arange(11), np.arange(6).reshape((3, 2))])
```

Comment:
It would be totally reasonable to make separate paths in these tests for versions of numpy older than 1.23. You would just need to add a line here: https://github.com/dask/dask/blob/5bbf072177fabdfc3421a5902aa23ee2bbf57ef4/dask/array/numpy_compat.py#L11-L12

and then follow the pattern in https://github.com/dask/dask/blob/5bbf072177fabdfc3421a5902aa23ee2bbf57ef4/dask/array/tests/test_reductions.py#L242-L243